packages/blade/src/components/Button/BaseButton/StyledBaseButton.native.tsx

```diff
@@ -101,7 +101,8 @@ const _StyledBaseButton: React.ForwardRefRenderFunction<BladeElementRef, StyledB
   );
 };
 
-const StyledBaseButton = React.forwardRef(_StyledBaseButton);
-StyledBaseButton.displayName = 'StyledBaseButton';
+const StyledBaseButton = /*#__PURE__*/ Object.assign(React.forwardRef(_StyledBaseButton), {
```

Comment:
So here, what is the main factor? 

The PURE comment or the Object.assign? which one does the magic?

Reply:
The PURE comment.

You can also define component inside a function and add prop inside that and return the component with imediate function call and this would probably work.
```ts
const StyledBaseButton = /*#__PURE__*/ (() => {
  const MyComponent = () => {};
  MyComponent.displayName = 'MyComponent';
  return MyComponent;
})();
```

The PURE comment only works on function calls so `Object.assign` just seems cleaner way compared to any other option.